From 59d0471a4127ffa446b8fa61d320f89af997da13 Mon Sep 17 00:00:00 2001
From: "Thulasimani,Sivakumar" <sivakumar.thulasimani@intel.com>
Date: Mon, 18 Jan 2016 17:35:22 +0530
Subject: [PATCH 08/19] SQUASHME! [VPG]: drm/i915: Fix upfront link training if
 MIPI is present

Using MIPI's CRTC for upfront link training during
resume results in MIPI display corrupted.
Upfront link training takes any available CRTC for
link training and it will take first CRTC (used by
MIPI) if we hotplug DP during sleep state and on
resume DPMS calls used to enable MIPI back
does not update local variables since there is no
modeset involved during resume, hence this patch
avoids reusing MIPI's crtc for upfront link training.

This patch should be merged to the following patch
Author:    Durgadoss R <durgadoss.r@intel.com>
Date:      Fri, 22 May 2015 14:30:07 +0530
Change-Id: I2ae866b32a0998abcf32826efa2504e203dc8ffa
Subject:   FOR_UPSTREAM [VPG]: drm/i915: Enable Upfront link
		training for type-C DP support

Change-Id: I74a2214251823251f1f21c8dd0171753550418eb
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-30099
Signed-off-by: Sivakumar Thulasimani <sivakumar.thulasimani@intel.com>
Reviewed-on: https://android.intel.com:443/498183
---
 drivers/gpu/drm/i915/intel_display.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 2b89fee..f0e259e 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -15242,6 +15242,9 @@ bool chv_upfront_link_train(struct drm_device *dev,
 		if (intel_crtc_active(&crtc->base))
 			continue;
 
+		if (intel_pipe_has_type(&crtc->base, INTEL_OUTPUT_DSI))
+			continue;
+
 		connector->new_encoder = encoder;
 		encoder->new_crtc = crtc;
 		encoder->base.crtc = &crtc->base;
-- 
1.9.1

